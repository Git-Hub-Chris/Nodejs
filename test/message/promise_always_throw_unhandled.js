'use strict';

require('../common');

// Check that the process will exit on the first unhandled rejection in case the
// environment variable is set to `'ERROR'`.

process.env.NODE_UNHANDLED_REJECTION = 'ERROR';

const ref1 = new Promise(() => {
  throw new Error('One');
});

const ref2 = Promise.reject(new Error('Two'));

// Keep the event loop alive to actually detect the unhandled rejection.
setTimeout(() => console.log(ref1, ref2), 1000);
