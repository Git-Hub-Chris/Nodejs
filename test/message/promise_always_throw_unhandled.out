*promises.js:*
          throw err;
          ^

Error: One
    at Promise (*promise_always_throw_unhandled.js:*:*)
    at new Promise (<anonymous>)
    at Object.<anonymous> (*promise_always_throw_unhandled.js:*:*)
    at *
    at *
    at *
    at *
    at *
    at *
    at *
